src/test/java/seedu/address/logic/commands/ListStudentCommandTest.java

```diff
@@ -27,13 +39,102 @@ public void setUp() {
     }
 
     @Test
-    public void execute_studentListIsNotFiltered_showsSameList() {
-        assertCommandSuccess(new ListStudentCommand(), model, ListStudentCommand.MESSAGE_SUCCESS, expectedModel);
+    public void execute_noIndexStudentListIsNotFiltered_showsSameList() {
+        assertCommandSuccess(new ListStudentCommand(), model,
+                ListStudentCommand.MESSAGE_LIST_ALL_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_noIndexStudentListIsFiltered_showsEverything() {
+        showStudentAtIndex(model, INDEX_FIRST_ITEM);
+        assertCommandSuccess(new ListStudentCommand(), model,
+                ListStudentCommand.MESSAGE_LIST_ALL_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_indexSpecifiedUnfilteredLists_showsStudentsInClass() {
+        ListStudentCommand listStudentCommand = new ListStudentCommand(INDEX_FIRST_ITEM);
+
+        ModuleClass specifiedClass = expectedModel.getFilteredModuleClassList().get(INDEX_FIRST_ITEM.getZeroBased());
+        Collection<UUID> expectedStudents = specifiedClass.getStudentUuids();
+        expectedModel.updateFilteredStudentList(new StudentInUuidCollectionPredicate(expectedStudents));
+
+        assertCommandSuccess(listStudentCommand, model,
+                String.format(ListStudentCommand.MESSAGE_LIST_CLASS_SPECIFIC_SUCCESS, specifiedClass), expectedModel);
     }
 
     @Test
-    public void execute_studentListIsFiltered_showsEverything() {
+    public void execute_indexSpecifiedStudentListIsFiltered_showsStudentsInClass() {
         showStudentAtIndex(model, INDEX_FIRST_ITEM);
-        assertCommandSuccess(new ListStudentCommand(), model, ListStudentCommand.MESSAGE_SUCCESS, expectedModel);
+        ListStudentCommand listStudentCommand = new ListStudentCommand(INDEX_FIRST_ITEM);
+
+        ModuleClass specifiedClass = expectedModel.getFilteredModuleClassList().get(INDEX_FIRST_ITEM.getZeroBased());
+        Collection<UUID> expectedStudents = specifiedClass.getStudentUuids();
+        expectedModel.updateFilteredStudentList(new StudentInUuidCollectionPredicate(expectedStudents));
+
+        assertCommandSuccess(listStudentCommand, model,
+                String.format(ListStudentCommand.MESSAGE_LIST_CLASS_SPECIFIC_SUCCESS, specifiedClass), expectedModel);
+    }
+
+    @Test
+    public void execute_indexSpecifiedClassListIsFiltered_showsStudentsInClass() {
+        showModuleClassAtIndex(model, INDEX_SECOND_ITEM);
+        showModuleClassAtIndex(expectedModel, INDEX_SECOND_ITEM);
+        ListStudentCommand listStudentCommand = new ListStudentCommand(INDEX_FIRST_ITEM);
+
+        ModuleClass specifiedClass = expectedModel.getFilteredModuleClassList().get(INDEX_FIRST_ITEM.getZeroBased());
+        Collection<UUID> expectedStudents = specifiedClass.getStudentUuids();
+        expectedModel.updateFilteredStudentList(new StudentInUuidCollectionPredicate(expectedStudents));
```

Comment:
I noticed that these 3 lines of code are repeated in above test cases. Perhaps can consider abstracting these out into a private method?